xio-core/src/main/java/com/xjeffrose/xio/application/XioServiceLocator.java

```diff
@@ -0,0 +1,27 @@
+package com.xjeffrose.xio.application;
+
+import com.xjeffrose.xio.core.ZkClient;
+import lombok.Getter;
+
+public class XioServiceLocator {
+  public static XioServiceLocator INSTANCE = null;
```

Comment:
we should add some access control to the INSTANCE so that only the application root scope can initialize it
... maybe add a private, empty constructor and perhaps a  static factory constructor which takes in the application bootstrap

Reply:
or maybe move this over to the bootstrap package and make the initialization "package private" in order to clearly communicate that this is intended as an ioc container that is initialized only at the application root

Reply:
fixed